python/filtering/filter_menu.py

```diff
@@ -351,6 +366,61 @@ def get_filters_container(self):
 
         return self.dock_widget if self.docked else self
 
+    def set_preset_filters(self, preset_filters, new_active_preset_filter_name=None):
+        """
+        Set the preset filters that can be toggled on/off in the menu.
+        The preset filters are stored as a dictionary where the key is the preset filter name,
+        and the value is the filter data in the format the SG API accepts.
+        :param preset_filters: dict of the preset filters to set.
+        :param new_active_preset_filter_name: string the name of the preset filter to activate if you want to change it.
+        """
+        # - If a new active preset filter name is provided, and it is different from the current
+        #   active preset filter name, set the new active preset filter name and emit the signal.
+        if new_active_preset_filter_name and self._active_preset_filter_name != new_active_preset_filter_name:
+            self._preset_filters = preset_filters
+            self.set_active_preset_filter(new_active_preset_filter_name)
+            return
+
+        # - If there is a current active preset filter name, and it is not in the new preset filters,
+        #   clear the active preset filter.
+        if self._active_preset_filter_name and self._active_preset_filter_name not in preset_filters:
+            self._preset_filters = preset_filters
+            self.set_active_preset_filter(None)
+            return
+
+        # - If there is a current active preset filter name, and it is in the new preset filters, but
+        #   the filters have changed, emit the signal.
+        if self._active_preset_filter_name:
+            # It is possible that the new preset_filters contains the same preset filter name as the
+            # active preset filter but the filters have changed.
+            # In this case, we should emit that the preset filter has changed.
+            active_preset_filter = self.get_active_preset_filter()
+            new_filter_matching_active_name = preset_filters[self._active_preset_filter_name]
+            if active_preset_filter != new_filter_matching_active_name:
+                self.preset_filter_changed.emit()
+            self._preset_filters = preset_filters
+            return
+
+        # No changes in active preset filter, just update the preset filters.
+        self._preset_filters = preset_filters
```

Comment:
I don't see this parameter being used (by tk-multi-shogunpanel), can we remove it to simplify this function logic?
If the caller wants to set the active preset filter, they should call `set_active_preset_filter`
```suggestion
    def set_preset_filters(self, preset_filters, new_active_preset_filter_name=None):
        """
        Set the preset filters that can be toggled on/off in the menu.
        The preset filters are stored as a dictionary where the key is the preset filter name,
        and the value is the filter data in the format the SG API accepts.
        :param preset_filters: dict of the preset filters to set.
        """

        self._preset_filters = preset_filters
```

Reply:
Yeah good call, I had implemented it for the loader, and maybe you are right I could have just called `set_active_preset_filter` after setting the filters. But since I didn't go through with implementing this feature in the loader I will remove the param as you suggest any way. Thanks!

That said it can't be completely simplified, because the preset filters could change from call to call. So I need to handle the situations where the active preset filter no longer exists, or it exists but has changed.